test/connectionStore.test.ts

```diff
@@ -216,6 +216,32 @@ suite('ConnectionStore tests', () => {
             }).catch(err => done(new Error(err)));
     });
 
+    test('RemoveProfile should not remove password from CredentialStore if keepCredentialStore is enabled', (done) => {
```

Comment:
It might be worthwhile adding another test for the keepCredentialStore = false case. This is implicitly covered by the removeProfile tests but it would be good to handle this here too.